massa-execution-exports/src/controller_traits.rs

```diff
@@ -23,8 +22,8 @@ pub trait ExecutionController: Send + Sync {
     /// * blockclique: new blockclique
     fn update_blockclique_status(
         &self,
-        finalized_blocks: Map<BlockId, Block>,
-        blockclique: Map<BlockId, Block>,
+        finalized_blocks: HashMap<Slot, BlockId>,
+        blockclique: HashMap<Slot, BlockId>,
```

Comment:
I believe it would be better to use `Map<BlockId, Slot>` instead to match the previous behaviour.

Reply:
I'm doing a branch based on this one to see if that was indeed the cause of https://github.com/massalabs/massa/issues/2469

Reply:
Be careful, we only use Maps when the key is already hashed (hence the prehashed trait), that's why when block id is the key it's ok to use a map, and it's not ok when slot is a key

Reply:
Yes I am aware of it as I mentioned in the 3rd point

Reply:
@Eitu33 Thanks for double checking. 

Re point 1: what "major behaviour change" does it create, and why is it "a risky thing to do"? I don't fully understand since even before the changes introduced in this PR the controller would end-up with a [`finalized_blocks: HashMap<Slot, BlockId>,`](https://github.com/massalabs/massa/pull/2428/files/c9dd61e7e54de844dae8fd2602aaa0a9af27b86b#diff-a78e41f7449ec5bc05f6f83364287a9ccc177125329251c608436baaf11c20c0) and a [`new_blockclique: Option<HashMap<Slot, BlockId>>`](https://github.com/massalabs/massa/pull/2428/files/c9dd61e7e54de844dae8fd2602aaa0a9af27b86b#diff-a78e41f7449ec5bc05f6f83364287a9ccc177125329251c608436baaf11c20c0R29)

Re point 2 and 3: yes that is correct, and since the controller ends up with a `HashMap<Slot, BlockId>` (and before the changes in this PR it is a `HashMap<Slot, (BlockId, Block)>`) anyway, why is that a problem? See https://github.com/massalabs/massa/pull/2428/files/c9dd61e7e54de844dae8fd2602aaa0a9af27b86b#diff-a78e41f7449ec5bc05f6f83364287a9ccc177125329251c608436baaf11c20c0L28

In both cases, the difference seems to be that in this PR the output is put directly in the message, as opposed to be mapped into it when receiving the message, hence I am wondering where the major change comes from.

Reply:
The changes I've made in https://github.com/massalabs/massa/pull/2473 did not fix https://github.com/massalabs/massa/issues/2469, I will be investigating this issue today.

Also I confirm that https://github.com/massalabs/massa/issues/2469 does not happen on main (since we weren't so sure).